binding/rdf4j/src/test/java/it/unibz/inf/ontop/rdf4j/repository/StrStartsFunctionTest.java

```diff
@@ -0,0 +1,189 @@
+package it.unibz.inf.ontop.rdf4j.repository;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Objects;
+
+import static org.junit.Assert.*;
+
+public class StrStartsFunctionTest extends AbstractRDF4JTest{
+    private static final String CREATE_DB_FILE = "/strstarts-function/sparql-strstarts.sql";
+    private static final String OBDA_FILE = "/strstarts-function/sparql-strstarts.obda";
+
+    @BeforeClass
+    public static void before() throws IOException, SQLException {
+        initOBDA(CREATE_DB_FILE, OBDA_FILE);
+    }
+
+    @AfterClass
+    public static void after() throws SQLException {
+        release();
+    }
+
+    @Test
+    public void testStrStartsConstants() {
+        String query = "PREFIX schema: <http://www.schema.org/>\n" +
+                "PREFIX sosa: <http://www.w3.org/ns/sosa/>\n" +
+                "SELECT ?s\n" +
+                "WHERE {\n" +
+                "?s a schema:Result.\n" +
+                "FILTER (STRSTARTS(\"name\", \"x\"))\n"+
+                "}";
+        ImmutableList<ImmutableMap<String, String>> result = executeQuery(query);
+        assertEquals(result, ImmutableList.of());
+    }
+
+    @Test
+    public void testStrStartsPredicate() {
+        String query = "PREFIX schema: <http://www.schema.org/>\n" +
+                "PREFIX sosa: <http://www.w3.org/ns/sosa/>\n" +
+                "SELECT DISTINCT ?s\n" +
+                "WHERE {\n" +
+                " ?s ?p ?o. \n" +
+                " ?s a schema:Person. \n" +
+                " FILTER (STRSTARTS(str(?p), \"http://www.schema.org/givenName\"))\n"+
+                "}";
+        // predicates are constants so they can be optimized
+        String reformulatedQuery = reformulateIntoNativeQuery(query).toLowerCase();
+        assertFalse(reformulatedQuery.contains("substring"));
+
+        ImmutableList<ImmutableMap<String, String>> result = executeQuery(query);
+        assertEquals(result, ImmutableList.of(
+                ImmutableMap.of("s", "http://www.schema.org/Person/1"),
```

Comment:
Could you please use a different prefix for the data? Schema.org, SOSA (in your PR description) are intended to be only for the ontology, not the data we map.